src/ui/map.js

```diff
@@ -1185,49 +1190,53 @@ class Map extends Camera {
         } else if (typeof projection === 'string') {
             projection = (({name: projection}: any): ProjectionSpecification);
         }
-        return this._updateProjection(projection);
+        this._explicitProjection = projection;
+        return this._updateProjection();
     }
 
-    _updateProjection(explicitProjection?: ProjectionSpecification | null): this {
-        const prevProjection = this.getProjection();
-        if (explicitProjection === null) {
-            this._explicitProjection = null;
+    _updateProjectionTransition() {
+        // The projection isn't globe, we can skip updating the transition
+        if (this.getProjection().name !== 'globe') {
+            return;
         }
-        const projection = explicitProjection || this.getProjection();
 
+        const tr = this.transform;
+        const projection = tr.projection.name;
         let projectionHasChanged;
-        // At high zoom on globe, set transform projection to Mercator while _explicitProjection stays globe.
-        if (projection && projection.name === 'globe' && this.transform.zoom >= GLOBE_ZOOM_THRESHOLD_MAX) {
-            projectionHasChanged = this.transform.setProjection({name: 'mercator'});
-            this.transform.mercatorFromTransition = true;
-        } else {
-            projectionHasChanged = this.transform.setProjection(projection);
-            this.transform.mercatorFromTransition = false;
+
+        if (projection === 'globe' && tr.zoom >= GLOBE_ZOOM_THRESHOLD_MAX) {
+            projectionHasChanged = tr.setMercatorFromTransition();
```

Comment:
If this condition is true`projectionHasChanged` will always be truthy, perhaps we can make the logic clearer by removing the function result
```suggestion
            tr.setMercatorFromTransition();
            projectionHasChanged = true;
```